core/transcoder_test.go

```diff
@@ -287,3 +287,31 @@ func TestTranscoder_Formats(t *testing.T) {
 	// sanity check the base format wasn't overwritten (has happened before!)
 	assert.Equal(ffmpeg.FormatNone, ffmpeg.P144p30fps16x9.Format)
 }
+
+func TestRecoverFromPanic(t *testing.T) {
+	assert := assert.New(t)
+
+	f := func() (err error) {
+		defer recoverFromPanic(&err)
+		panic(struct{}{})
+	}
+
+	err := f()
+
+	assert.NotNil(err)
+	assert.Equal("unrecoverable transcoding failure", err.Error())
```

Comment:
Should we also add an assertion to make sure that the "unrecoverable transcoding failure" error is also a `UnrecoverableError`?

Reply:
Good idea, added :+1: